macrotranslator.py

```diff
@@ -0,0 +1,87 @@
+from macros import MacroMap, Macro, Invocation
+from translationconfig import TranslationConfig
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+class MacroTranslator:
+    def __init__(self, translation_config: TranslationConfig) -> None:
+        self.translation_config = translation_config
+
+    def generate_macro_translations(self,
+                                    mm: MacroMap) -> dict[Macro, str | None]:
```

Comment:
I think it's more idiomatic replace all these `| None` types with [`Optional`](https://docs.python.org/3/library/typing.html#typing.Optional)s

```suggestion
                                    mm: MacroMap) -> dict[Macro, Optional[str]]:
```

You'll need to add `from typing import Optional` to the top of the file as well.

Reply:
AFAIK it's [encouraged](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/6) to use the "T | None" syntax though Optional isn't deprecated. 

I personally find it a little clearer to understand in this context because it's clear that a macro either has a string translation or None.

Reply:
Ah I wasn't aware of that; interesting. So I suppose `|` is preferable over `Optional` in return types as well argument types? In any case, if you prefer the union syntax then keep it since you are the main developer on the translator at the moment.